File: model-client/src/main/java/org/modelix/model/api/IConcept.java
package org.modelix.model.api;

public interface IConcept {
  boolean isSubconceptOf(IConcept superConcept);
  boolean isExactly(IConcept concept);

  Iterable<IProperty> getProperties();
  Iterable<IChildLink> getChildLinks();
  Iterable<IReferenceLink> getReferenceLinks();

  IProperty getProperty(String name);
  IChildLink getChildLink(String name);
  IReferenceLink getReferenceLink(String name);
}

File: model-client/src/main/kotlin/org/modelix/model/api/IConcept.kt
package org.modelix.model.api

interface IConcept {
    fun isSubconceptOf(superConcept: IConcept?): Boolean
    fun isExactly(concept: IConcept?): Boolean
    val properties: Iterable<IProperty?>?
    val childLinks: Iterable<IChildLink?>?
    val referenceLinks: Iterable<IReferenceLink?>?
    fun getProperty(name: String?): IProperty?
    fun getChildLink(name: String?): IChildLink?
    fun getReferenceLink(name: String?): IReferenceLink?
}